Assets/Scripts/Game/UserInterface/ListBox.cs

```diff
@@ -367,17 +373,56 @@ private void DecideTextColor(TextLabel label, int i)
             }
             else if (i == highlightedIndex)
             {
-                label.TextColor = listItems[i].highlightedTextColor;
+                label.TextColor = listItems[i].Enabled ?
+                    listItems[i].highlightedTextColor
+                    : listItems[i].highlightedDisabledTextColor;
                 label.ShadowPosition = shadowPosition;
                 label.ShadowColor = listItems[i].shadowColor;
             }
             else
             {
-                label.TextColor = listItems[i].textColor;
+                label.TextColor = listItems[i].Enabled ?
+                    listItems[i].textColor
+                    : listItems[i].disabledTextColor;
                 label.ShadowPosition = shadowPosition;
                 label.ShadowColor = listItems[i].shadowColor;
             }
         }
+
+        private int FindNextEnabled(int currentIndex, bool previous)
```

Comment:
I feel like this could have been split into 2 functions, there's barely any common code between the two branches

Reply:
Sure!